### DISCUSSION TITLE
User-Roles as Entity.

### DISCUSSION THREAD

**Discussion by chris576** [symfony/symfony#52988] [Q&A] [answered]
Hello there.

I would like to save my Userroles as Entities wihtin the Database. So instead of returning an array of strings `['ROLE_ADMIN', 'ROLE_USER']` in the `public function getRoles() : array` function in `User` Entity, I'd like to return an array of entites. 

This is what I have done yesterday so far. But now I am not able to identify a User to a Role. For example 
`$secruity->isGranted('ROLE_USER')`, nor `$secruity->isGranted(new Role('ROLE_User'))` do return true.

**Comment by stof** [comment, marked as answer]
Well, the method of the interface must return an array of strings. That's the contract defined in the interface.
You can store them as entity if you want but the getter defined in the interface must still return strings.
